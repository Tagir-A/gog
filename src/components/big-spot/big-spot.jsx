import React, { Component } from "react";
import "./big-spot.css";

class BigSpot extends Component {
  render() {
    return (
      <section className="big-spot">
        <h3 className="big-spot__title">GAME OF THE WEEK</h3>
        <button className="btn big-spot__btn" />
      </section>
    );
  }
}

export { BigSpot };
