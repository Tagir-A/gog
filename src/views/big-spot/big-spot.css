.big-spot {
  margin-bottom: 20px;
}

.big-spot__btn {
  display: block;
  width: 100%;
  padding-bottom: 34.63%;
  background: url("../../assets/game-spot@2x.png") no-repeat center;
  background-size: contain;
}
