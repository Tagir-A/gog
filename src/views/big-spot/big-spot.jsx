import React, { Component } from "react";
import "./big-spot.css";

function BigSpot() {
  return (
    <section className="big-spot">
      <h3 className="big-spot__title">GAME OF THE WEEK</h3>
      <button className="btn big-spot__btn" />
    </section>
  );
}

export { BigSpot };
