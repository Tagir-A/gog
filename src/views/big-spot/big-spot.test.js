import React from "react";
import { shallow } from "enzyme";
import { BigSpot } from "./big-spot";
import { wrap } from "module";

it("renders without crashing", () => {
  const wrapper = shallow(<BigSpot />);
  expect(wrapper.html()).toMatchSnapshot();
});

it("contains title with proper text", () => {
  const wrapper = shallow(<BigSpot />);
  expect(wrapper.find(".big-spot__title").exists()).toBeTruthy();
  expect(wrapper.find(".big-spot__title").text()).toEqual("GAME OF THE WEEK");
});
