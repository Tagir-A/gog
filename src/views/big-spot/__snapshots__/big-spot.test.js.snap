// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `"<section class=\\"big-spot\\"><h3 class=\\"big-spot__title\\">GAME OF THE WEEK</h3><button class=\\"btn big-spot__btn\\"></button></section>"`;
